drivers/tee/optee/optee_bench.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2014, Linaro Limited
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ */
+
+#ifndef OPTEE_BENCH_H
+#define OPTEE_BENCH_H
+
+/* max amount of timestamps */
+#define TEE_BENCH_MAX_STAMPS	100
+#define TEE_BENCH_RB_SIZE sizeof(struct tee_ringbuf) \
+		+ sizeof(struct tee_time_st) * TEE_BENCH_MAX_STAMPS
+#define TEE_BENCH_DEF_PARAM		3
+
+/* OP-TEE susbsystems ids */
+#define TEE_BENCH_KMOD		0x00000002
+
+
+/* storing timestamps */
+struct tee_time_st {
+	u64 cnt;		/* stores value from CNTPCT register */
+	u64 addr;		/* stores value from program counter register */
+	u64 src; 			/* OP-TEE subsystem id */
+};
+
+/* memory layout for shared memory, where timestamps will be stored */
+struct tee_ringbuf {
+	u64 tm_ind;		/* index of the last timestamp in stamps[] */
+	struct tee_time_st stamps[];
+};
+
+
+
+#ifdef CONFIG_OPTEE_BENCHMARK
+
+/* Program counter */
+#define TEE_BENCH_PC(src) \
+	asm volatile("mov %0, r15": "=r"(src));
```

Comment:
Here also would be better as a function, but obviously it has to be always inlined to be correct. `__attribute__((always_inline)) inline` should do the trick I suppose.
Even if you keep a macro you should make it look like: `pc = read_pc();` rather than `TEE_BENCH_PC(pc);` because it's more readable in my opinion.